python/feathub/online_stores/redis_client.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Feathub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, List
+
+import pandas as pd
+import redis
+
+from feathub.common.utils import (
+    deserialize_object_with_protobuf,
+)
+from feathub.online_stores.online_store_client import OnlineStoreClient
+from feathub.processors.flink.table_builder.redis_utils import serialize_and_join_keys
+from feathub.table.schema import Schema
+
+
+class RedisClient(OnlineStoreClient):
+    """
+    An online store client that reads feature values from Redis.
+    """
+
+    def __init__(
+        self,
+        schema: Schema,
+        host: str,
+        port: int = 6379,
+        username: str = None,
+        password: str = None,
+        db_num: int = 0,
+        namespace: str = "default",
+        keys: Optional[List[str]] = None,
+        timestamp_field: Optional[str] = None,
+    ):
+        super().__init__()
+        self.namespace = namespace
+
+        self.key_names = keys
+        self.key_types = [schema.get_field_type(x) for x in self.key_names]
+
+        self.feature_names = [
```

Comment:
Maybe rename it as `all_feature_names` to differentiate it from the `feature_names` parameter passed to `get(...)`?